### PR TITLE
.x: SPAWN NO HOME OPTION

### PR DESCRIPTION
**Pull request by darbyjack**

### CODE REVIEW

**Review by Ichbinjoe — DISMISSED** [review]

File: Essentials/src/config.yml [hunk 1]
```diff
@@ -920,6 +920,9 @@ spawn-join-listener-priority: high
 # When users die, should they respawn at their first home or bed, instead of the spawnpoint?
 respawn-at-home: false
 
+# If no home is set, would you like to send the player to spawn?
```
Comment (mdcfe) [inline Essentials/src/config.yml]: This is a bit unclear - it should be obvious that this only applies to `/home`.

**Review by mdcfe — CHANGES_REQUESTED** [review]

File: Essentials/src/com/earth2me/essentials/Settings.java [hunk 2]
```diff
@@ -1623,4 +1623,9 @@ private double _getMaxProjectileSpeed() {
     public double getMaxProjectileSpeed() {
         return maxProjectileSpeed;
     }
+
+    @Override
+    public boolean isSpawnIfNoHome() {
+        return config.getBoolean("spawn-if-no-home", false);
```
Comment (mdcfe) [inline Essentials/src/com/earth2me/essentials/Settings.java]: This should default to true to preserve the previous behaviour.

File: Essentials/src/config.yml [hunk 3]
```diff
@@ -920,6 +920,9 @@ spawn-join-listener-priority: high
 # When users die, should they respawn at their first home or bed, instead of the spawnpoint?
 respawn-at-home: false
 
+# If no home is set, would you like to send the player to spawn? (Only applies to /home)
+spawn-if-no-home: false
```
Comment (mdcfe) [inline Essentials/src/config.yml]: Similarly this should default to true.

**Review by mdcfe — CHANGES_REQUESTED** [review]

File: Essentials/src/com/earth2me/essentials/Settings.java (hunk 2 above)
Comment (darbyjack) [reply Essentials/src/com/earth2me/essentials/Settings.java]: Updated.

File: Essentials/src/config.yml (hunk 3 above)
Comment (darbyjack) [reply Essentials/src/config.yml]: Updated.

**Review by mdcfe — APPROVED** [review]

**Comment by mdcfe** [comment]
This has been implemented in [EssentialsX 2.17.1.19 and above](https://essentialsx.cf/downloads.html).

**Pull request merged**